tests/test_boards.py

```diff
@@ -1295,6 +1297,15 @@ def test_energenie(mock_factory):
         pins[3].assert_states_and_times([(0.0, True), (0.0, True)])
         pins[4].assert_states_and_times([(0.0, False)])
         pins[5].assert_states_and_times([(0.0, False), (0.1, True), (0.25, False)])
+        [pin.clear_states() for pin in pins]
+        device3.on()
+        assert device3.value
+        pins[0].assert_states_and_times([(0.0, False), (0.0, True)])
+        pins[1].assert_states_and_times([(0.0, True), (0.0, False)])
+        pins[2].assert_states_and_times([(0.0, True), (0.0, False)])
+        pins[3].assert_states_and_times([(0.0, True), (0.0, False)])
+        pins[4].assert_states_and_times([(0.0, False)])
+        pins[5].assert_states_and_times([(0.0, False), (0.1, True), (0.25, False)])
```

Comment:
Would you mind adding tests that `device3.value = 0` and `device3.value = 1` work as expected, and `device3.value = None` raises an exception? Thanks!